frontend/src/modules/controllers/ValueClicker.tsx

```diff
@@ -61,18 +83,34 @@ export const ValueClicker = ({
   const [isRotaryActive, setIsRotaryActive] = React.useState(false);
   const [activeRef, setActiveRef] = React.useState<string | null>();
 
+  /**
+   * Triggers callback with updated value
+   *
+   * @param {number} step - Alarm step difference
+   *
+   * @returns callback function
+   */
   const update = (step: number) => {
     const change = value + step;
     setDisableIncrement(change >= max);
     setDisableDecrement(change <= min);
     return onClick(change);
   };
 
+  /**
+   * Validating & disabling button when reached min/max value
```

Comment:
What's the relationship, if any, between this `useEffect` and the `update` function defined above? It looks like the `update` function is doing everything this `useEffect` does, and more?

Reply:
`update` function executes only when user clicks on buttons. On initializing `ValueClicker` component, parent component can provide `value` in props. In that case `update` will not be executed so we use Effects on Component initialization and add the same validation.